eng/pipelines/templates/steps/vcpkg.yml

```
@@ -54,16 +126,32 @@ steps:
         # then pull
         git reset --hard HEAD
         git pull origin master
+        git checkout ${{ parameters.VcpkgVersion }}
 
         ./bootstrap-vcpkg.sh
-    condition: and(succeeded(), not(contains(variables['OSVmImage'], 'macOS')), not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')))
-    displayName: Upgrade vcpkg
+    condition: >-
+      and(
+        succeeded(),
+        ne(variables['VcpkgRestoredFromCache'], 'true'),
+        not(contains(variables['OSVmImage'], 'macOS')),
+        not(eq(variables['${{ parameters.DependenciesVariableName }}'], ''))
+      )
+    displayName: Non-Mac - Set vcpkg version and bootstrap
```

nit: keep the displayName matching with the other above `MacOS & Linux - vcpkg bootstrap` (alternatively, change the above to be `MacOS & Linux - Set vcpkg version and bootstrap`
```suggestion
    displayName: Non-MacOS - vcpkg bootstrap
```